src/runtime/SandBox.cpp

```diff
@@ -37,7 +37,11 @@ SandBox::SandBoxResult SandBox::run(const std::function<Value()>& scriptRunner)
         result.msgStr = result.result.toString(state);
     } catch (const Value& err) {
         result.error = err;
-        result.msgStr = result.error.toString(state);
+
+        ErrorObject* errAsErrorObject = err.asObject()->asErrorObject();
+        ObjectPropertyName name(state.context()->staticStrings().message);
+        ObjectGetResult prop = errAsErrorObject->getOwnProperty(state, name);
+        result.msgStr = prop.m_value.asString();
```

Comment:
Existing code base seems to avoid unnecessary temp variables even if that results in longer code lines. (E.g., see src/runtime/GlobalObjectBuiltinError.cpp:33)
Moreover, digging into the private fields of an object of another class is usually bad design / nasty / dangerous (and/or). In this case all of these apply as `m_value` is in a union.
So, this code could be rewritten as:

```cpp
result.msgStr = err.asObject()->asErrorObject()->getOwnProperty(state, state.context()->staticStrings().message).value(state, err).asString();
```

Which would avoid temp vars and private fields.

BUT! I'm not convinced that this is the right approach, because there are quite some asserts hidden in this call chain like landmines.
- Is it sure that `err` is an (error) object? JavaScript allows throwing strings, numbers, bools as well.
- Is it sure that the result returned by `getOwnProperty` (for the property "message") has a value?
- Is it sure that the value is a string?

(The latter two questions are typical targets for fuzzing attacks. JavaScript allows the redefinition of several properties and it is easy to run into assertion failures when it turns out that some assumptions don't hold. Actually, this PR aims at fixing a similar assumption.)

So, I wonder whether `result.msgStr = result.error.toString(state);` should rather be wrapped into another try-catch instead?
```suggestion
        try {
            result.msgStr = result.error.toString(state);
        } catch (const Value& err) {
            result.msgStr = String::emptyString;
        }
```

Reply:
The proposed try-catch solution will work, but what should `result.msgStr` be set to in the catch block? Using `String::emptyString` will not give us informative error messages, and it can potentially interfere with other errors. For example:
```js
Error.prototype.toString = Object.getOwnPropertyDescriptor.toString;
try {
  throw new ReferenceError();
} catch (e) {
  console.log(e)
}
```
This script will catch the thrown `ReferenceError`, but when we try to `console.log()`, it will throw another `ReferenceError` because there is no `console.log()` in Escargot. But we won't be able to see that from the error message. This is what it displayed for my example:
```
Uncaught :
test.js (6:3)
function toString() { [native function] }  (-1:-1)
```
It can be quite hard to tell where the problem is from an error message like this if the user doesn't know that there is no `console.log()`, and this is just a trivial example.
Somehow we should get the error message from `result.error` to solve this, but I'm not really sure how that could be possible, since we can't use `toString()` in this catch block and my workaround-ish approach has the problems you described. Maybe we can leave it as `String::emptyString` as you suggested for now until there is a better solution. What do you think?

Reply:
Something like "error while executing script but couldn't convert error to string" could also work with me but I'm not sure how more informative that would be. So, I'm good with either an empty string or with some pre-set message. (Feel free to reword the text as you see fit if you opt for the non-empty string.)